routing/payment_session.go

```diff
@@ -194,13 +194,28 @@ type paymentSession struct {
 
 	// log is a payment session-specific logger.
 	log btclog.Logger
+
+	// A transformation which will be applied to every route created for
+	// this payment session.
+	routeTransform RouteTransformFunc
+}
+
+// Option is a functional option for configuring the payment session.
+type Option func(*paymentSession)
+
+// WithRouteTransform sets a route transformation function.
+func WithRouteTransform(routeTransform RouteTransformFunc) Option {
+	return func(ps *paymentSession) {
+		ps.routeTransform = routeTransform
+	}
```

Comment:
These two symbols are used only inside the package, so they can be made private.

IIUC, the only publicly visible change in this commit is new field `routing.SessionSource.RouteTransform`.

Reply:
Updated ✅